conn.go

```diff
@@ -4,23 +4,36 @@ import (
 	"fmt"
 	"io"
 	"net"
+	"time"
 )
 
 type transmitFunc func(buf []byte) (bool, error)
 
 type Conn struct {
-	addr     net.Addr
-	conn     net.PacketConn
-	protocol *Protocol
+	addr net.Addr
+	conn net.PacketConn
+
+	p *Protocol
+
```

Comment:
warning: exported function NewConn should have comment or be unexported (golint)